packages/story-editor/src/app/media/uploadQueue/index.ts

```diff
@@ -644,12 +701,12 @@ function useMediaUploadQueue() {
      * @type {Array} In-progress items.
      */
     const progress = state.queue.filter(
-      (item) =>
+      (item: QueueItem) =>
         ![
-          ITEM_STATUS.PENDING,
-          ITEM_STATUS.CANCELLED,
-          ITEM_STATUS.UPLOADED,
-          ITEM_STATUS.FINISHED,
+          ItemStatus.Pending,
+          ItemStatus.Cancelled,
+          ItemStatus.Uploaded,
+          ItemStatus.Finished,
```

Comment:
The next many lines here can be greatly simplified by introducing a tiny util like:

```ts

    const hasStatus =
      (...states: ItemStatus[]) =>
      (item: QueueItem) =>
        states.includes(item.state);
```

The arrays and functions become respectively:

```ts
    const pending = state.queue.filter(hasStatus(ItemStatus.Pending));
    ...
    const isTranscoding = state.queue.some(hasStatus(ItemStatus.Transcoding));
```

At least I think that's a lot cleaner.

The more complex checks can be filtered first using this as:

```ts
    const isCurrentResourceTranscoding = (resourceId: ResourceId) =>
      state.queue
        .filter(hasStatus(ItemStatus.Transcoding))
        .some((item) => item.resource.id === resourceId);
```

Reply:
Fixed.